docs/workflows/best-practices-architecture.md

```diff
@@ -0,0 +1,76 @@
+
+Many systems use a combination of these reasons when designing their component architecture. For example, it is quite common to define the top-level architecture as a number of microservices that are deployed separately. But every microservice also has an internal architecture, which is mainly based on dividing the codebase into logical areas. Such a combination is not "wrong". It is a perfectly reasonable choice which indicates that modularizing a system needs to consider both technical and human aspects.
+
+These considerations will change over time. That is perfectly normal and healthy. One of your component architecture's goals is to facilitate architecture evolution. It is not humanly possible to foresee all possible future changes, so it's normal for your architecture to change over time in order to meet those changing needs. That means designing your component architecture is not a one-off activity, like many other software engineering activities it's an iterative process.
+{: .attention }
+
+## What is the best way to modularize your codebase?
+
+This is where the dreaded "it depends" answer comes into play. The goal of your component structure is to communicate the intended structure of your system to people. Such a goal makes it very hard to give generic best practices on what your components should be, since this depends on the functionality of your system. 
+
+People therefore generally find it easier to define a component structure based on technical layers. This provides a generic structure that can be used for any type of system regardless of its functionality. 
+
+A commonly used example of such a reference architecture is the controller-service-repository pattern. As the name implies, this pattern defines three layers:
+
+- **Repositories** act as the data access layer. The other layers are not allowed to access the database directly. 
+- **Services** contain all business logic. They access the repositories whenever they require data access.
+- **Controllers** act as input/output for accessing the business logic. Controllers can only access the service layer, they cannot directly access the repository layer.
+
+Again, note these definitions are purely technical and do not depend on the functionality of the system. Systems that are using this pattern always use the following modularization:
+
+<img src="../images/methodology/components-example-1.png" width="500" />
+
+Defining components in this way is perfectly fine. Using a reference modularization pattern also has a major benefit: Anyone who knows the pattern will immediately understand the modularization of your system. People will understand where new functionality should be implemented, since every layer has clearly defined boundaries. 
```

Comment:
"Defining components in this way is perfectly fine." --> remove this sentence? it is not always the best/perfectly fine, after all